src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
So, considering that I still think having short paths is desired to make things work seamlessly on Windows, considering also that [truncating a sha256 looks fine](http://security.stackexchange.com/questions/34796/truncating-the-output-of-sha256-to-128-bits) for the use case, I replaced md5 by a truncated sha256. Case insensitivity is here equivalent to truncating a bit more the hash on Windows, which seems safe also for the use case (see previous comment). OK for you?